week1/index.js

```diff
@@ -1,10 +1,71 @@
-// To run this file: node index.js
-// Then in your browser: http://localhost:8080
-var http = require('http');
+// Declare global variables
+const http = require('http');
+//start server 
+const server = http.createServer();
+// the state which will be increased or decreased on the user request
+let state = 10; 
+//This port was already chosen by the assignment maker.
+const port = 8080;
+//Object containing the respons text and the content type.
+let responsToSendBack = {
+    code:'<html></html>',
+    type:'text/html',
+    statusCode: 200
+};
 
-var port = 8080;
 
-var server = http.createServer();
+// Function that checks which URL input we got from the user 
+//Initiates the right action based on the user choice
+function checkInput (url) {
+    switch (url) {
+        case '/state':
+            stateResponsHtml();
+            break;
+        case '/add':
+            addOne();
+            operationsResponseHtml();
+            console.log(state);
+            break;
+        case '/remove':
+            subOne();
+            operationsResponseHtml();
+            console.log(state);
+            break;
+        case '/reset':
+            state = 10;
+            operationsResponseHtml();
+            break;
+        default:
+            otherUrls();
+    }
+}
+
+//Function to increase the state by 1
+function addOne (){
```

Comment:
These functions take no parameters, yet modify global variables. Globals can be quite dangerous, especially in async programming.

What you probably want to do here is write functions which take arguments and return a modified output, so something more like:

```
function addOne (n) {
    return n++;
}
```
This is probably small enough functionality that you don't need a function for this, and could just use `++` inline.